File: Python/Trabalho_POO_parte_2/Bicicleta.py
from Veiculo import *

class Bicicleta(Veiculo):
    def __init__(self, id: int, rodas: int):
        super().__init__(id, rodas)
        # tipo para facilitar na hora de identificar o tipo de veiculo, dava pra fazer se essa variavel mais ia ficar muita gambiarra e seria dificil explicar
        self.__tipo = "B"
        print(f"Bicicleta com ID {self.get_id()} foi construida")

    # metodo get
    def get_tipo(self) -> int:
        return self.__tipo

    # funções abstrata herdada pelo pai veiculo
    def mover(self):
        if self.verificar_pneu():
            new_distancia = self.get_distancia_percorrida() + 2
            self.set_distancia_percorrida(new_distancia)
            print(f"Bicicleta - ID:{self.get_id()} andou +2 blocos")
        else: 
            print(f"Bicicleta com ID {self.get_id()} não possui todos os pneus calibrados")

    def desenhar(self):
        print(" "*self.get_distancia_percorrida() + "   __o")
        print(" "*self.get_distancia_percorrida() + " _`\<,_:")
        print(" "*self.get_distancia_percorrida() + "(*)/ (*)")

File: C++/Trabalho_POO_parte_2/Bicicleta.cpp
#include "Bicicleta.h"

Bicicleta::Bicicleta(int id, int rodas) : Veiculo(id, rodas)
{
    tipo = 'B';
    std::cout << "Bicicleta com ID " << get_id() << " foi construída" << endl;
}

char Bicicleta::get_tipo() 
{
    return tipo;

}
void Bicicleta::mover() 
{
    if (verificar_pneu())
    {
        int new_distancia = get_distancia_percorrida() + 2;
        set_distancia_percorrida(new_distancia);
        std::cout << "Bicicleta - ID:" << get_id() << " andou +2 blocos" << endl;
    }
    else
        std::cout << "Bicicleta com ID " << get_id() << " não possui todos os pneus calibrados" << endl;
}

void Bicicleta::desenhar() 
{
    string espacos(get_distancia_percorrida(), ' ');
    std::cout << espacos << "   __o" << endl;
    std::cout << espacos << " _`\\<,_: " << endl;
    std::cout << espacos << " (*)/ (*)" << endl;
}